cpp/benchmarks/search/search.cpp

```diff
@@ -14,63 +14,45 @@
  * limitations under the License.
  */
 
+#include <benchmarks/common/generate_input.hpp>
+#include <benchmarks/fixture/benchmark_fixture.hpp>
+#include <benchmarks/synchronization/synchronization.hpp>
+
+#include <cudf_test/base_fixture.hpp>
 #include <cudf_test/column_wrapper.hpp>
 
+#include <cudf/filling.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
 #include <cudf/search.hpp>
 #include <cudf/sorting.hpp>
 #include <cudf/types.hpp>
 
-#include <benchmark/benchmark.h>
-
 #include <random>
 
-#include <benchmarks/fixture/benchmark_fixture.hpp>
-#include <benchmarks/synchronization/synchronization.hpp>
-#include <cudf_test/base_fixture.hpp>
-
 class Search : public cudf::benchmark {
 };
 
-auto make_validity_iter()
-{
-  static constexpr int r_min = 1;
-  static constexpr int r_max = 10;
-
-  cudf::test::UniformRandomGenerator<uint8_t> rand_gen(r_min, r_max);
-  uint8_t mod_base = rand_gen.generate();
-  return cudf::detail::make_counting_transform_iterator(
-    0, [mod_base](auto row) { return (row % mod_base) > 0; });
-}
-
 void BM_column(benchmark::State& state, bool nulls)
 {
-  const cudf::size_type column_size{(cudf::size_type)state.range(0)};
-  const cudf::size_type values_size = column_size;
-
-  auto col_data_it = cudf::detail::make_counting_transform_iterator(
-    0, [=](cudf::size_type row) { return static_cast<float>(row); });
-  auto val_data_it = cudf::detail::make_counting_transform_iterator(
-    0, [=](cudf::size_type row) { return static_cast<float>(values_size - row); });
-
-  auto column = [&]() {
-    return nulls ? cudf::test::fixed_width_column_wrapper<float>(
-                     col_data_it, col_data_it + column_size, make_validity_iter())
-                 : cudf::test::fixed_width_column_wrapper<float>(col_data_it,
-                                                                 col_data_it + column_size);
-  }();
-  auto values = [&]() {
-    return nulls ? cudf::test::fixed_width_column_wrapper<float>(
-                     val_data_it, val_data_it + values_size, make_validity_iter())
-                 : cudf::test::fixed_width_column_wrapper<float>(val_data_it,
-                                                                 val_data_it + values_size);
-  }();
-
-  auto data_table = cudf::sort(cudf::table_view({column}));
+  auto const column_size{static_cast<cudf::size_type>(state.range(0))};
+  auto const values_size = column_size;
+
+  auto init_data  = cudf::make_fixed_width_scalar<float>(static_cast<float>(0));
+  auto init_value = cudf::make_fixed_width_scalar<float>(static_cast<float>(values_size));
+  auto step       = cudf::make_fixed_width_scalar<float>(static_cast<float>(-1));
+  auto column     = cudf::sequence(column_size, *init_data);
+  auto values     = cudf::sequence(values_size, *init_value, *step);
+  if (nulls) {
+    column->set_null_mask(create_random_null_mask(column->size(), 0.1, 1).first);
+    values->set_null_mask(create_random_null_mask(values->size(), 0.1, 2).first);
+  }
```

Comment:
I suspect that this benchmark will include the time to compute a null count. I know in some benchmarks, we do not want to include that time. Should we use a structured binding to get both the mask and the null count from `create_random_null_mask` and call `set_null_count` as well?
```suggestion
  if (nulls) {
    auto [column_null_mask, column_null_count] = create_random_null_mask(column->size(), 0.1, 1);
    column->set_null_mask(column_null_mask);
    column->set_null_count(column_null_count);
    auto [values_null_mask, values_null_count] = create_random_null_mask(values->size(), 0.1, 1);
    values->set_null_mask(values_null_mask);
    values->set_null_count(values_null_count);
  }
```

Reply:
updated